src/control/deleteCollection.ts

```diff
@@ -1,22 +1,20 @@
 import { ManageIndexesApi } from '../pinecone-generated-ts-fetch/control';
-import { buildConfigValidator } from '../validator';
-import { CollectionNameSchema } from './types';
 import type { CollectionName } from './types';
+import { PineconeArgumentError } from '../errors';
 
 /**
  * The name of collection to delete.
  */
 export type DeleteCollectionOptions = CollectionName;
 
 export const deleteCollection = (api: ManageIndexesApi) => {
-  const validator = buildConfigValidator(
-    CollectionNameSchema,
-    'deleteCollection'
-  );
-
   return async (collectionName: DeleteCollectionOptions): Promise<void> => {
-    validator(collectionName);
-
+    if (!collectionName || collectionName.length === 0) {
+      throw new PineconeArgumentError(
+        'You must pass a non-empty string for `collectionName` in order to delete a' +
+          ' collection'
```

Comment:
nit: Personal preference but I really hate these multi-line strings and would prefer to avoid them if possible. Is this a prettier or linting thing?

Reply:
It's prettier, I think! But I can try to shorten them